phylogeny/prototypeSelection.py

```diff
@@ -125,3 +125,82 @@ def prototype_selection_exhaustive(dm: DistanceMatrix, num_prototypes: int,
         if d > max_dist:
             max_dist, max_set = d, s
     return max_set
+
+
+def prototype_selection_constructive_maxdist(dm, num_prototypes, steps=10):
+    '''Heuristically select k prototypes for given distance matrix.
+       Greedily grow the set of prototypes by adding the element with the
+       largest sum of distances to the non-prototype elements.
+
+    Parameters
+    ----------
+    dm: skbio.stats.distance.DistanceMatrix
+        pairwise distances for all elements in the full set S.
+        Must be symmetric and non-hollow.
+    num_prototypes: int
+        Number of prototypes to select for distance matrix.
+        Must be >= 2, since a single prototype is useless.
+        Must be smaller than the number of elements in the distance matrix,
+        otherwise no reduction is necessary.
+
+    Returns
+    -------
+    sequence of str
```

Comment:
as input, it would be appropriate to describe as "Iterable of str" which suggests `list`, `tuple`, `generator`, etc would work. But in this case, this is describing the return type, which presumably is specific?